.github/workflows/ci.yml

```
@@ -6,76 +6,133 @@ on:
+        with:
+          go-version-file: go.mod
+          cache: true
+
+      - name: Test with race detector & coverage
+        run: |
+          go test -race -count=1 -coverprofile=coverage.out -covermode=atomic ./...
+
+      - name: Publish coverage summary
+        if: always()
+        shell: bash
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:12:49: Double quote to prevent globbing and word splitting

(shellcheck)